python/ray/tune/integration/xgboost.py

```diff
@@ -81,15 +86,24 @@ class _TuneCheckpointCallback(TuneCallback):
     Args:
         filename (str): Filename of the checkpoint within the checkpoint
             directory. Defaults to "checkpoint".
+        frequency (int): How often to save checkpoints. Per default, a
+            checkpoint is saved every five iterations.
 
     """
 
-    def __init__(self, filename: str = "checkpoint"):
+    def __init__(self, filename: str = "checkpoint", frequency: int = 5):
         self._filename = filename
+        self._frequency = frequency
 
-    def __call__(self, env):
+    @staticmethod
```

Comment:
why does this have to be a static method?

Reply:
Hmm, looks like it might not need to be, but I'll merge first as it's not too consequential.

Reply:
The idea here is that in xgboost_ray we'll send this callback to the driver via the queue. A non-static method will pickle `self` when it's being sent via the object store, while a static method shouldn't do so.